lib/libutils/isoc/sprintf.c

```diff
@@ -17,3 +18,16 @@ int sprintf(char *str, const char *fmt, ...)
 
 	return retval;
 }
+
+int __sprintf_chk(char *str, int flag __unused, size_t slen,
+		  const char *fmt, ...)
+{
+	int retval;
+	va_list ap;
+
+	va_start(ap, fmt);
+	retval = __vsnprintf(str, slen, fmt, ap, false);
```

Comment:
http://refspecs.linux-foundation.org/LSB_4.1.0/LSB-Core-generic/LSB-Core-generic/libc---sprintf-chk-1.html

mentions a few checks that should be performed. Checking for slen == 0 or that retval > = slen shouldn't be too hard.

Reply:
> Checking for slen == 0 or that retval > = slen shouldn't be too hard.

Although I did see a check for `slen == 0` in glibc implementation [here](https://sourceware.org/git/?p=glibc.git;a=blob;f=debug/sprintf_chk.c;h=c4444f7ff8dc64a1d95f2270b281ceadbc36fa65;hb=HEAD) but I couldn't find check for `retval > = slen`. Should we just return `-1` for these condition checks?

Reply:
The spec seems to suggest that we should panic instead.

Reply:
Added corresponding checks.